plugins/modules/idrac_server_config_profile.py

```diff
@@ -755,83 +752,178 @@ def preview_scp_redfish(module, idrac, http_share, import_job_wait=False):
-            share["file_name"] = module.params.get("scp_file")
-        buffer_text = None
+        share, _scp_file_name_format = get_scp_share_details(module)
+        share["file_name"] = module.params.get("scp_file")
+        buffer_text = get_buffer_text(module, share)
         share_dict = share
         if share["share_type"] == "LOCAL":
-            file_path = "{0}{1}{2}".format(share["share_name"], os.sep, share["file_name"])
-            if not exists(file_path):
-                module.fail_json(msg=INVALID_FILE)
-            with open(file_path, "r") as file_obj:
-                buffer_text = file_obj.read()
             share_dict = {}
         module.params["job_wait"] = job_wait
-        scp_response = idrac.import_scp_share(shutdown_type=module.params["shutdown_type"],
-                                              host_powerstate=module.params["end_host_power_state"],
-                                              job_wait=module.params["job_wait"],
-                                              target=scp_targets,
-                                              import_buffer=buffer_text, share=share_dict, )
-        job_id = scp_response.headers["Location"].split("/")[-1]
-        if module.params["job_wait"]:
-            job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(
-                idrac, iDRAC_JOB_URI.format(job_id=job_id))
-            scp_response = job_dict
+        idrac_import_scp_params = {
+            "import_buffer": buffer_text, "target": scp_targets, "share": share_dict, "job_wait": module.params["job_wait"],
+            "host_powerstate": module.params["end_host_power_state"], "shutdown_type": module.params["shutdown_type"]
+        }
+        scp_response = idrac_import_scp_share(module, idrac, idrac_import_scp_params)
     else:
         scp_response = idrac.import_scp(import_buffer=import_buffer, target=scp_targets, job_wait=module.params["job_wait"])
     scp_response = response_format_change(scp_response, module.params, share.get("file_name"))
-    if isinstance(scp_response, dict) and scp_response.get("TaskStatus") == "Critical":
-        module.fail_json(msg="Failed to {0} scp.".format(command), scp_status=scp_response)
+    exit_on_failure(module, scp_response, command)
     return scp_response
 
 
+def idrac_import_scp_share(module, idrac, job_params):
+    scp_response = idrac.import_scp_share(**job_params)
+    job_id = scp_response.headers["Location"].split("/")[-1]
+    if module.params["job_wait"]:
+        job_failed, _msg, job_dict, _wait_time = idrac_redfish_job_tracking(
+            idrac, iDRAC_JOB_URI.format(job_id=job_id))
+        if job_failed:
+            module.exit_json(failed=True, status_msg=job_dict, job_id=job_id)
+        scp_response = job_dict
+    return scp_response
+
+
+def validate_input(module, scp_components):
+    if len(scp_components) != 1 and "ALL" in scp_components:
+        module.fail_json(msg=SCP_ALL_ERR_MSG)
```

Comment:
Avoid using fail_json throughout the code, use exit_json with failed=True

Reply:
i dont see a change in behavior of both

Reply:
When we use fail_json we get traceback error, which we don't want.

Reply:
currently it fails with the msg